conf/gulp-tasks/library.gulpfile.js

```diff
@@ -1,46 +1,86 @@
+const LocalFileParser = require('../i18n/localfileparser');
+const TranslationResolver = require('../i18n/translationresolver');
+const Translator = require('../i18n/translator');
 
-const bundleTaskFactory = new BundleTaskFactory(getLibraryVersion());
-
-function bundle () {
-  return bundleTaskFactory.create(BundleType.MODERN)();
-}
+/**
+ * Creates the un-minified legacy JS bundle, compiles CSS, and kicks off the watch task.
+ * @returns {Promise<Function>}
+ */
+exports.dev = function devJSBundle () {
+  return createTaskFactory(DEV_LOCALE).then(devTaskFactory => {
+    return new Promise(resolve => {
+      return parallel(
+        series(devTaskFactory.create(BundleType.LEGACY_IIFE), watchJS(devTaskFactory)),
+        series(compileCSS, watchCSS)
+      );
+    });
+  });
+};
 
-function legacyBundleIIFE () {
-  return bundleTaskFactory.create(BundleType.LEGACY_IIFE)();
-}
+/**
+ * Creates a build task per locale, and combines them into a parallel task.
+ * @returns {Promise<Function>}
+ */
+exports.default = function defaultJSBundle () {
+  const localizedTaskPromises = BUILD_LOCALES.map(locale => {
+    return createTaskFactory(locale).then(createBundles);
+  });
+  return Promise.all(localizedTaskPromises).then(localizedTasks => {
+    return new Promise(resolve => parallel(...localizedTasks)(resolve));
+  });
+};
 
-function legacyBundleUMD () {
-  return bundleTaskFactory.create(BundleType.LEGACY_UMD)();
+/**
+ * Creates modern, legacy and legacyUMD bundles in parallel.
+ *
+ * @param {BundleTaskFactory} bundleTaskFactory
+ */
+async function createBundles (bundleTaskFactory) {
+  return parallel(
+    series(
+      bundleTaskFactory.create(BundleType.MODERN),
+      bundleTaskFactory.minify(BundleType.MODERN)
+    ),
+    series(
+      bundleTaskFactory.create(BundleType.LEGACY_IIFE),
+      bundleTaskFactory.minify(BundleType.LEGACY_IIFE)
+    ),
+    series(
+      bundleTaskFactory.create(BundleType.LEGACY_UMD),
+      bundleTaskFactory.minify(BundleType.LEGACY_UMD)
+    ),
+    series(compileCSS)
+  );
 }
 
-function minifyJS () {
-  return src('./dist/answers-modern.js')
-    .pipe(rename('answers-modern.min.js'))
-    .pipe(uglify())
-    .pipe(dest('dist'));
-}
+/**
+ * Creates a BundleTaskFactory configured with translations for the given locale.
+ *
+ * @param {string} locale
+ * @returns {BundleTaskFactory} bundleTaskFactory
+ */
+async function createTaskFactory (locale) {
+  const localFileParser = new LocalFileParser(path.join(__dirname, '../i18n/translations'));
+  const translation = await localFileParser.fetch(locale);
 
-function minifyLegacy () {
-  return src('./dist/answers.js')
-    .pipe(rename('answers.min.js'))
-    .pipe(uglify())
-    .pipe(dest('dist'));
-}
+  const translator = await Translator.create(locale, [], { [locale]: { translation } });
+  const translationResolver = new TranslationResolver(
+    translator,
+    (translationResult, interpValues, count) => {
```

Comment:
In some cases, translationResult will already be a format string (if there's interpolation but no pluralization). Does this runtime call generator handle that case?

Reply:
Yes! I added a test to the PR description:
```js
const test3 = replaceWithTranslation({
  phrase: 'I am looking for my child named [[name]]',
  interpolationValues: {
    name: this._config.title,
  }
});
```

is changed to:
```js
var test3 = ANSWERS.translateJS("I am looking for my child named [[name]]", {name:this._config.title}, undefined);
```